### DISCUSSION TITLE
Text file based Pipelines

### DISCUSSION THREAD

**Discussion by miwojc** [elyra-ai/elyra#2293] [Q&A]
Is it possible to define pipeline in text file rather than in visual editor?
Advantage of that is that it will be possible to do version control on pipeline settings.

**Comment by ptitzler** [comment]
Hi @miwojc, the `.pipeline` file format is not documented and changes over time. It's therefore not advisable to manually create or update those files. Can you clarify what pipeline settings you are referring to?

**Reply by miwojc** [reply]
So like when you create a pipeline / flow visually connecting jupyter notebooks in visual editor.

**Reply by ptitzler** [reply]
Sorry, this fell through the cracks. There is no programmatic support for authoring `.pipeline` files that we support.
